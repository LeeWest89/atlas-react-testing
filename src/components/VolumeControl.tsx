import { useState } from "react";
import volume from "../assets/volume.svg";

export default function VolumeControl () {
  const [volLevel, setVolLevel] = useState(75);

  const handleVol = (event: React.ChangeEvent<HTMLInputElement>) => {
    setVolLevel(Number(event.target.value));
  };

  return (
    <div className="flex flex-row max-w-[42.9rem] md:max-w-[25rem] h-[1.5rem] items-center">
      <img src={volume} alt="volume icon" className="w-[1.1664375rem] h-[1.25rem]" />
      <input type="range" min="0" max="100" value={volLevel} onChange={handleVol} className="accent-gray-500 ml-2 w-full" />
    </div>
  );
}