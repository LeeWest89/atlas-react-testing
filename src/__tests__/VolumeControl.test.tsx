import { render, screen, fireEvent } from "@testing-library/react";
import { expect, test } from "vitest";
import VolumeControl from "../components/VolumeControl";

// Test if VolumeControl renders
test("VolumeControl renders", () => {
  const { container } = render(<VolumeControl />)
  expect(container).toMatchSnapshot();
});

// Test if volume level starts at 75
test("VolumeControl renders", () => {
  const inputElement = screen.getByRole('slider') as HTMLInputElement;
  expect(inputElement.value).toBe('75');
});

// Test if volume level changes on slide
test("VolumeControl updates", () => {
  const inputElement = screen.getByRole('slider') as HTMLInputElement;
  fireEvent.change(inputElement, { target: { value: '25' } });
  expect(inputElement.value).toBe('25');
});

// Test if volume level changes on slide outside slider min range
test("VolumeControl updates even outside min", () => {
  const inputElement = screen.getByRole('slider') as HTMLInputElement;
  fireEvent.change(inputElement, { target: { value: '-25' } });
  expect(inputElement.value).toBe('0');
});

// Test if volume level changes on slide outside slider max range
test("VolumeControl updates even outside max", () => {
  const inputElement = screen.getByRole('slider') as HTMLInputElement;
  fireEvent.change(inputElement, { target: { value: '123' } });
  expect(inputElement.value).toBe('100');
});
