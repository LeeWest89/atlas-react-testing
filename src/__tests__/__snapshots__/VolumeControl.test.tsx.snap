// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl renders 1`] = `
<div>
  <div
    class="flex flex-row max-w-[42.9rem] md:max-w-[25rem] h-[1.5rem] items-center"
  >
    <img
      alt="volume icon"
      class="w-[1.1664375rem] h-[1.25rem]"
      src="/src/assets/volume.svg"
    />
    <input
      class="bg-lightning-blue ml-2 w-full"
      max="100"
      min="0"
      type="range"
      value="75"
    />
  </div>
</div>
`;
